feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c

```diff
@@ -356,22 +358,49 @@ static void vif_state_custom_options_get(struct schema_Wifi_VIF_State *vstate,
 							buf);
 			}
 		} else if (strcmp(opt, "rts_threshold") == 0) {
-                        if (tb[WIF_ATTR_RTS_THRESHOLD]) {
-                                buf = blobmsg_get_string(tb[WIF_ATTR_RTS_THRESHOLD]);
-                                set_custom_option_state(vstate, &index,
-                                                        custom_options_table[i],
-                                                        buf);
-                        }
-                } else if (strcmp(opt, "dtim_period") == 0) {
-                        if (tb[WIF_ATTR_DTIM_PERIOD]) {
-                                buf = blobmsg_get_string(tb[WIF_ATTR_DTIM_PERIOD]);
-                                set_custom_option_state(vstate, &index,
-                                                        custom_options_table[i],
-                                                        buf);
-                        }
-                }
+			if (tb[WIF_ATTR_RTS_THRESHOLD]) {
+				buf = blobmsg_get_string(tb[WIF_ATTR_RTS_THRESHOLD]);
+				set_custom_option_state(vstate, &index,
+						custom_options_table[i],
+						buf);
+			}
+		} else if (strcmp(opt, "dtim_period") == 0) {
+			if (tb[WIF_ATTR_DTIM_PERIOD]) {
+				buf = blobmsg_get_string(tb[WIF_ATTR_DTIM_PERIOD]);
+				set_custom_option_state(vstate, &index,
+						custom_options_table[i],
+						buf);
+			}
+		}
+
+	}
+}
 
+
+#define unnamed_sec "\
+		#/bin/bash \n\
+		cfgid=$( uci show dhcp.@dnsmasq[0] ) \n\
+		cfgid=${cfgid%%'='*} \n\
+		cfgid=${cfgid#*'.'} \n\
+		echo $cfgid \n\
+		"
+
+int get_unnamed_section(void) {
```

Comment:
I dont think this a good idea. why do the section have to be unnamed ? we had a mail discussion about this and I believe you should auto generate unique names

Reply:
I need  to write dns names under dnsmasq unnamed section, so I used "@dnsmasq[0]" to write. but when I try to to get the info I've added, back to state table the same name "@dnsmasq[0]" is not working. since the config ID is generating automatically. I just read that respective name and used while updating to the state.

Reply:
ok, lets just patch the dnsmasq package and give the section a name, shall I push such a patch and you can rework this code based on the new named section ?

Reply:
sure, I'll do that, thank you.

Reply:
pushed the dnsmasq patch to pending